src/modules/ekf2/EKF/mag_control.cpp

```diff
@@ -304,20 +303,6 @@ void Ekf::checkMagDeclRequired()
 	_control_status.flags.mag_dec = (_control_status.flags.mag_3D && (not_using_ne_aiding || user_selected));
 }
 
-void Ekf::checkMagInhibition()
-{
-	_is_yaw_fusion_inhibited = shouldInhibitMag();
-
-	if (!_is_yaw_fusion_inhibited) {
-		_mag_use_not_inhibit_us = _imu_sample_delayed.time_us;
-	}
-
-	// If magnetometer use has been inhibited continuously then a yaw reset is required for a valid heading
```

Comment:
How is this replaced? Would the mag fusion stop by itself after 5 sec of inhibited fusion and then reset when restarting?

Reply:
I believe it only matters for this special case here.

https://github.com/PX4/PX4-Autopilot/blob/88b5130eab6da1e37df9b0f976505743762db425/src/modules/ekf2/EKF/mag_control.cpp#L228-L237

Reply:
Does this capture the same overall intent? Reset gyro z bias if we haven't had any valid heading fusion for 5 seconds?

Maybe it would make slightly more sense directly in fuseYaw() or resetQuatStateYaw()?

Reply:
I think that's fine. The forced reset after 5s of mag inhibition wasn't a great thing anyway.